docs/how-tos/use_the_api.rst

```diff
@@ -0,0 +1,86 @@
+How To Use the REST API
+#######################
+
+.. How-tos should have a short introduction sentence that captures the user's goal and introduces the steps.
+
+This how-to will help get setup to be able to make aunthenticated requests to
+the edx-platform REST API.
+
+Assumptions
+***********
+
+.. This section should contain a bulleted list of assumptions you have of the
+   person who is following the How-to.  The assumptions may link to other
+   how-tos if possible.
+
+* You have access to the edx-platform Django Admin (``/admin``) Panel.
+
+* You have a user that you want to make the rest calls as (``UserA``).
+
+* You are familiar with `the basics of HTTP and Rest`_
+
+* For the purposes of this tutorial we'll assume your LMS is located at
+  https://lms.example.com
+
+.. _the basics of HTTP and Rest: https://code.tutsplus.com/a-beginners-guide-to-http-and-rest--net-16340t
+
+Steps
+*****
+
+.. A task should have 3 - 7 steps.  Tasks with more should be broken down into digestible chunks.
+
+#. Go to https://lms.example.com/admin/oauth2_provider/application/
+
+#. Click :guilabel:`Add Application`
+
+#. Choose "UserA" for the user.
+
+#. Choose ``Confidential`` Client Type
+
+#. Choose "Client Credentials" for the Authorization Grant Type
+
+#. Set a name for your application.
+
+#. Save the ``client_id`` and ``client_secret``.
+
+#. The best way to interact with the edx-platform REST API is by making
+   requests using the JWT Authorization header.  User the ``client_id`` and
+   ``client_secret`` to get a JWT token.
+
+   .. code-block:: python
+
+      import base64
+      import requests
+
+      client_id = "vovj0AItd9EnrOKjkDli0HpSF9HoooaTY9yueafn"
+      client_secret = "a3Fkwr24dfDSlIXt3v3q4Ob41CYQNZyGmtK8Y8ax0srpIa2vJON3OC5Rvj1i1wizsIUv1W1qM1Q2XPeuyjucNixsHXZsuw1dn2B9nH3IyjSvuFb5KoydDvWX8Hx8znqD"
+
+      credential = f"{client_id}:{client_secret}"
+      encoded_credential = base64.b64encode(credential.encode("utf-8")).decode("utf-8")
+
+      headers = {"Authorization": f"Basic {encoded_credential}", "Cache-Control": "no-cache"}
+      data = {"grant_type": "client_credentials", "token_type": "jwt"}
```

Comment:
I think using the `client_id` as a param with public clients is fine but using the `client_secret` in the query params exposes it to various HTTP logs and so it can leak more easily in production environments.  The Authorization header should be within the TLS transaction so it should be more secure.

Reply:
I'm planning on removing `BasicAuthentication` as part of https://github.com/openedx/edx-platform/issues/33028. Do your instructions in-fact rely on `BasicAuthentication`? Should we make it an explicit auth class on `/access_token`, or does this work in some other way? As I said, I've never done this before so I wasn't aware it was an option.

Reply:
I dug into it, the reason it's working is because the views behind `/access_token` are just django views and not DRF views.  They go from oauth_dispatch to  DOT and don't use DRF to do manage permissions.